### PR TITLE
Add authentication on mobile

### PR DESCRIPTION
**Pull request by Losdrew**

### CODE REVIEW

**Review by firtap — APPROVED** [review]

File: ScanShop.Mobile/ScanShop.Mobile/Services/ConfigurationService.cs
Comment (Wednumi) [inline ScanShop.Mobile/ScanShop.Mobile/Services/ConfigurationService.cs:1]: why did you not use native configuration IConfiguration something like
var config = new ConfigurationBuilder().AddJsonFile(path).Build();
serviceCollection.AddTransient<IConfiguration>(provider => config);
?

File: ScanShop.Mobile/ScanShop.Mobile/Services/HttpClientService.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            var configurationService = DependencyService.Get<IConfigurationService>();
+            _httpClient.BaseAddress = configurationService.GetBaseUrlAsync().Result;
+        }
+
+        public void SetBearerToken(string token)
+        {
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        }
+
+        public async Task<HttpResponseMessage> PostAsync<T>(string endpoint, T payload)
```
Comment (Wednumi) [inline ScanShop.Mobile/ScanShop.Mobile/Services/HttpClientService.cs:30]: why do you need to create additional method like this when PostAsJsonAsync in IHttpClient exist?

File: ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
+
             MainPage = new AppShell();
+
+            if (string.IsNullOrEmpty(SecureStorage.GetAsync("BearerToken").Result))
+            {
+                Shell.Current.GoToAsync("//LoginPage");
+            }
+            else
+            {
+                var bearerToken = SecureStorage.GetAsync("BearerToken");
```
Comment (Wednumi) [inline ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs:23]: firstly, you call SecureStorage.GetAsync("BearerToken"); twice. Secondly, I think it is better to brought setting bearer into httpClientService, for example in InitializeAsync method with injecting IConfiguration there

File: ScanShop.Mobile/ScanShop.Mobile/Services/ConfigurationService.cs
Comment (Losdrew) [reply ScanShop.Mobile/ScanShop.Mobile/Services/ConfigurationService.cs:1]: It is native to ASP.NET, but it isn't available in Xamarin Forms

File: ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs (hunk 2 above)
Comment (Losdrew) [reply ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs]: Fixed in new commit

File: ScanShop.Mobile/ScanShop.Mobile/Services/HttpClientService.cs (hunk 1 above)
Comment (Losdrew) [reply ScanShop.Mobile/ScanShop.Mobile/Services/HttpClientService.cs:41]: Fixed in new commit

File: ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs [hunk 3]
```diff
@@ -14,17 +13,9 @@ public App()
...
-            }
-            else
-            {
-                var bearerToken = SecureStorage.GetAsync("BearerToken");
-                var httpClientService = DependencyService.Get<IHttpClientService>();
-                httpClientService.SetBearerToken(bearerToken.Result);
-                Shell.Current.GoToAsync("//OrdersPage");
-            }
+            var httpClientService = DependencyService.Get<IHttpClientService>();
+            httpClientService.InitializeAsync();
```
Comment (Wednumi) [inline ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs:17]: Never leave Async method unfinished if it is not supposed to. Use synchronous .Wait() or .Result

Comment (Losdrew) [reply ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs]: Fixed in new commit

**Review by Wednumi — APPROVED** [review]

**Pull request merged**